docs/specification/Feb2020Ballot.md

```diff
@@ -0,0 +1,534 @@
+```
+HTTP/1.1 202 Accepted
+
+wss://hub.example.com/ee30d3b9-1558-464f-a299-cbad6f8135de
+```
+
+### Subscription Denial
+
+If (and when) the subscription is denied, the Hub SHALL inform the subscriber. This can occur when the subscription is requested for a variety of reasons, or it can occur after the subscription had already been accepted because the Hub no longer supports that subscription (e.g. it has expired). The communication mechanism for a subscription denial varies per `hub.channel.type`, but the information communicated from the Hub to the subscriber does not. 
+
+
+Field | Optionality | Type | Description
+--- | --- | --- | ---
+`hub.mode` | Required | *string* | The literal string "denied".
+`hub.topic` | Required | *string* | The topic given in the corresponding subscription request. MAY be a guid.
+`hub.events` | Required | *string* | A comma-separated list of events from the Event Catalog corresponding to the events string given in the corresponding subscription request, which are being denied. 
+`hub.reason` | Optional | *string* | The Hub may include a reason. The subscription MAY be denied by the Hub at any point (even if it was previously accepted). The Subscriber SHOULD then consider that the subscription is not possible anymore.
+
+
+The below [webhook flow diagram](https://drive.google.com/file/d/1Z7Z7mw0f_gm8lqdBJcwqQV8MD9PnVhQs/view?usp=sharing) and [websocket flow diagram](https://drive.google.com/file/d/1uByH7aDXBXINlexFIn7TT2oCnjf_Ae4e/view?usp=sharing) and examples illustrate the subscription denial sequence and message details.
+
+#### `webhook` Subscription Denial
+
+To deny a subscription with `hub.channel.type`=`webhook`, the Hub sends an HTTP GET request to the subscriber's callback URL as given in the subscription request.  This request appends the fields as query string arguments. The subscriber SHALL respond with an HTTP success (2xx) code.
+
+
+###### `webhook` Subscription Denial Sequence
+![Subscription denial flow diagram](/img/Denied%20Webhook%20Subscription%20Sequence.png)
+
+###### `webhook` Subscription Denial Example
+```
+GET https://app.example.com/session/callback/v7tfwuk17a?hub.mode=denied&hub.topic=fdb2f928-5546-4f52-87a0-0648e9ded065hub.events=patient-open,patient-close&hub.reason=session+unexpectedly+stopped HTTP 1.1
+Host: subscriber
+```
+
+#### `websocket` Subscription Denial
+To deny a subscription with `hub.channel.type`=`websocket`, the Hub sends a json object to the subscriber through the established websocket connection. 
+
+###### `websocket`Subscription Denial Sequence
+
+![Subscription denial flow diagram](/img/Denied%20Websocket%20Subscription%20Sequence.png)
+
+###### `websocket` Subscription Denial Example 
+```
+{
+   "hub.mode": "denied",
+   "hub.topic":" "fba7b1e2-53e9-40aa-883a-2af57ab4e2c",
+   "hub.events": "patient-open,patient-close",
+   "hub.reason": "session unexpectedly stopped"
+}
+```
+
+### `webhook` Intent Verification
```

Comment:
There isn't really an intent verification in websockets but it may be worth adding a note of that. I was looking for websockets information under the intent verification section (even though there really isn't anything) so I'm sure other people would too.

Reply:
Will, not this section already exists:

> ### `webhook` Intent Verification
If a subscription with `hub.channel.type`=`webhook` is not denied, the Hub SHALL perform the verification of intent of the subscriber, this applies to apps unsubscribing as well. The `hub.callback` url verification process ensures that the subscriber actually controls the callback url. Note the verification of intent and the `hub.callback` url SHALL only be used for subscriptions over webhooks.

Any suggestions for improving it?

Reply:
Ahh got it. That is probably fine, I guess if we were to add a line that is a little more explicit in saying "websockets does not have any intent verification process" or something similar then that couldn't hurt.

Reply:
Will, I just added a "subscription confirmation" step, see: #299 and changes in this PR? What do you think?